src/CommunityToolkit/Xamarin.CommunityToolkit/Helpers/LocalizationResourceManager.shared.cs

```diff
@@ -11,30 +12,40 @@ public class LocalizationResourceManager : INotifyPropertyChanged
 		public static LocalizationResourceManager Current { get; } = new LocalizationResourceManager();
 
 		ResourceManager resourceManager;
+		CultureInfo currentCulture = Thread.CurrentThread.CurrentUICulture;
 
-		public void Init(ResourceManager resource)
+		public void Init(ResourceManager resource) =>
+			resourceManager = resource;
+
+		public void Init(ResourceManager resource, CultureInfo initialCulture)
 		{
 			resourceManager = resource;
+			CurrentCulture = initialCulture;
 		}
 
-		public string this[string text] => resourceManager.GetString(text, CurrentCulture);
+		public string GetValue(string text) =>
+			resourceManager.GetString(text, CurrentCulture);
 
-		public void SetCulture(CultureInfo language)
-		{
-			Thread.CurrentThread.CurrentUICulture = language;
-			Invalidate();
-		}
+		public string this[string text] =>
+			GetValue(text);
 
-		public string GetValue(string text) => resourceManager.GetString(text, CultureInfo.CurrentCulture);
+		[Obsolete("Use " + nameof(CurrentCulture) + " to set culture")]
+		public void SetCulture(CultureInfo language) => CurrentCulture = language;
 
-		public CultureInfo CurrentCulture => Thread.CurrentThread.CurrentUICulture;
+		public CultureInfo CurrentCulture
+		{
+			get => currentCulture;
+			set
+			{
+				currentCulture = value;
+				Invalidate();
+			}
+		}
```

Comment:
I have a question about this change... Is this change something in the usage? If no I would say to keep the prop `private set` and use the `SetCulture` method to handle it. 

I know that is good to use the C# features, but this isn't a good reason to Obsolete an API.

Reply:
Makes sense. Just figured since package is new and this method is generally not used much, it's still time to move to using property instead. It's not only good use of C#, it's how everything that updates culture does it (`CultureInfo.CurrentCulture`, `Thread.CurrentThread.CurrentUICulture`, `AppResources.Culture`).

Are you sure it's worth sticking to this method? I really feel like property is the way to go here (especially considering that we a moving to backing field, so this method no longer has any external side effects)

Reply:
@maxkoshevoi if this change isn't the fix for the bug, I would say to keep at it is (with method instead of Prop) and it's just to avoid to make the API Obsolete. Anyway, we need the input of other maintainers before. 
If we agree to Obsolete this API this will be an API change and should target the develop branch.

@AndreiMisiukevich , @sthewissen , @jfversluis , @jsuarezruiz, what do you think about it?

Reply:
@pictos This change isn't the fix for the bug. This fix is sort of a blocker since `LocalizationResourceManager` cannot be reliably used until it's merged, so we can split this PR into two. One for fixing the bug and second for deprecating the `SetCulture` method.

Reply:
@maxkoshevoi  that sounds good! Before opening a PR to deprecate the method, can you open an issue first?

Reply:
@pictos Removed `[Obsolete]` attribute from `SetMethod` and made `CurrentCulture` setter private. I'll create an issue to reverse that later today,

Reply:
@maxkoshevoi Thank you so much! For the help!